plugins/modules/cloud/misc/proxmox_kvm.py

```diff
@@ -966,7 +966,7 @@ def create_vm(self, vmid, newid, node, name, memory, cpu, cores, sockets, update
         if 'agent' in kwargs:
             try:
                 # The API also allows booleans instead of e.g. `enabled=1` for backward-compatibility.
-                kwargs['agent'] = boolean(kwargs['agent'], strict=True)
+                kwargs['agent'] = '1' if boolean(kwargs['agent'], strict=True) else '0'
```

Comment:
If this is correct, then why did passing `True` and `False` (not the strings, but the Python values) to proxmoxer work before 0be7b6e7b91bb2d7b6e16773cc9b93a655ca0dd6?

Reply:
#5194 is another bug. The problem is that the module is sending a string to the proxmox api that won't work. There are two options.

1. Send real booleans
2. Send `1` or `0` as string

As far as I can see the most compatible way is to send `1` or `0` as string, so I implemented it that way in the fix above.

Reply:
But right now, it **is** sending real booleans if the code path is used that you are modifying. So I don't see how your fix chnages anything.

Reply:
I don't know where exactly in the code that happens but the boolean somewhere gets transformed to a string containing the value `True`. Maybe a fix can also be to deactivate that conversion.

Reply:
@felixfontein 
Hello! I have faced this issue too. Before 5.5.0 `agent` was `bool` and was parsed around line 895 (Sanitize kwargs). In result it had value `int` 0 or 1 which then passed to Proxmox API successfully.
To save old behavior for booleans I would make this value integer (same method that used in 'sanitizing')
```suggestion
                kwargs['agent'] = int(boolean(kwargs['agent'], strict=True))
```

Reply:
Ah, that makes sense. Yes, int `int()` should definitely be used here.

Also this PR needs a [changelog fragment](https://docs.ansible.com/ansible/devel/community/collection_development_process.html#creating-a-changelog-fragment) before it can be merged.